batch/batch/driver/pool.py

```diff
@@ -251,6 +252,13 @@ async def create_instances(self):
                 # parallelism will be bounded by thread pool
                 await asyncio.gather(*[self.create_instance() for _ in range(instances_needed)])
 
+        if self.scheduler.ci_no_viable_instances.value > 5 and self.live_free_cores_mcpu_by_zone[GCP_ZONE] == 0:
```

Comment:
I feel like the right fix is to retrieve ready cores by user, and then check here:
```
if ready_cores_mcpu_by_user['ci'] > 0 and self.live_free_cores_mcpu_by_zone[GCP_ZONE] == 0:
   ...
```